File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages_data/RefPackage.java
package pl.edu.mimuw.loxim.protocol.packages_data;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class RefPackage extends Package {
		public final static long ID=(long)134;
	
    	/**
**/
    	private java.math.BigInteger valueId=java.math.BigInteger.ZERO;
    	
	public RefPackage()
	{
	}

		public RefPackage(
		    	java.math.BigInteger a_valueId    	)
	{
        	valueId=a_valueId;
    
	}
		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
				try{
	    			valueId=reader.readUint64();
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
				try{
	    			writer.writeUint64(valueId);		
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}	
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		RefPackage p=(RefPackage)obj;
	    if (!(((valueId==p.getValueId())||((valueId!=null)&&(p.getValueId()!=null)&&((valueId).equals(p.getValueId())))))) 
return false;
				return true;
	}
	
//================= GETTERS AND SETTERS ===================

        	public java.math.BigInteger getValueId(){return valueId;};
		
    	public void setValueId(java.math.BigInteger a_valueId){ valueId=a_valueId;};
		
    
	 
}

File: Protogen/loxim_protocol/cpp/protocol/packages_data/RefPackage.cpp
#include <string.h>
#include <stdlib.h>
#include "RefPackage.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

RefPackage::RefPackage(\
		uint64_t a_valueId
		)
{
	valueId=a_valueId;
}

RefPackage::RefPackage()
{
	valueId=0;
}

RefPackage::~RefPackage()
{
}

void RefPackage::serializeW(PackageBufferWriter *writer)
{
	Package::serializeW(writer);
	writer->writeUint64(valueId);
}

bool RefPackage::deserializeR(PackageBufferReader *reader)
{
	if (!Package::deserializeR(reader)) return false;
	valueId=reader->readUint64();
	return true;
}

bool RefPackage::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!Package::equals(_package)) return false;
	RefPackage* p=(RefPackage*)_package;
	if (!(getValueId()==p->getValueId())) return false;
	return true;
}
